io/disk/vfc-tests/vfc_unmap_map.py

```diff
@@ -0,0 +1,337 @@
+        for line in output:
+            if vfchost in line:
+                return line.split(":")[-1]
+        return None
+
+    def get_vfc_client(self, vfchost, vios_name):
+        '''
+        returns the linux_name of corresponding drc_name
+        '''
+        cmd = 'viosvrcmd -m %s -p %s -c "lsmap -all -npiv -field  name\
+               vfcclient -fmt :"' % (self.server, vios_name)
+        output = self.run_command(cmd)
+        self.log.info("output value : %s" % output)
+        for line in output:
+            if vfchost in line:
+                return line.split(":")[-1]
+        return None
+
+    def get_vfchost_status(self, vfchost, vios_name):
+        '''
+        returns the linux_name of corresponding drc_name
+        '''
+        cmd = 'viosvrcmd -m %s -p %s -c "lsmap -all -npiv -field  name\
+               Status -fmt :"' % (self.server, vios_name)
+        output = self.run_command(cmd)
+        self.log.info("output value : %s" % output)
+        for line in output:
+            if vfchost in line:
+                return line.split(":")[-1]
+        return None
+
+    def get_paths(self, vfc_client):
+        '''
+        returns the mpaths corresponding to linux name
+        '''
+        paths = []
+        cmd = "ls -l /sys/block/"
+        output = process.system_output(cmd).decode('utf-8').splitlines()
+        for line in output:
+            if "/%s/" % vfc_client in line:
+                paths.append(line.split("/")[-1])
+        return paths
+
+    def vfc_client_status_verify(self, operation, vfc_client):
+        '''
+        Returns the WWPNs of give client slot number
+        '''
+        err_host = []
+        cmd = 'cat /sys/class/fc_host/host2/port_state'
+        status = process.system_output(cmd).decode('utf-8')
+        if operation == "map":
+            if status != "online":
+                err_host.append(vfc_client)
+        elif operation == "unmap":
+            if status != "Linkdown":
+                err_host.append(vfc_client)
+        if err_host:
+            self.fail("operation: %s client status verify failed, \
+                       status:%s" % (operation, status))
+        else:
+            self.log.info("operation: %s client status verfy success, \
+                           status:%s" % (operation, status))
+
+    def vfchost_map_unmap(self, operation, vfc_dic):
+        '''
+        Adds and removes a Network virtualized device based
+        on the operation
+        '''
+        if operation == 'map':
+            cmd = 'vfcmap -vadapter %s -fcp %s' \
+                   % (vfc_dic["vfchost"], vfc_dic["fcs_name"])
+        else:
+            cmd = 'vfcmap -vadapter %s -fcp %s' % (vfc_dic["vfchost"])
+        try:
+            self.run_command(cmd)
+        except CommandFailed as cmdfail:
+            self.log.debug(str(cmdfail))
+            self.fail("vfchost %s operation failed" % operation)
+        time.sleep(self.opp_sleep_time)
```

Comment:
why do we need sleep here ?

Reply:
before verification need some sleep time for changes to  reflections like status of different elements. so this is required.

Reply:
can't we have some check in sysfs instead of sleep ?